lib/LedgerEventStorage.js

```diff
@@ -159,10 +159,16 @@ class LedgerEventStorage {
           if(!database.isDuplicateError(e)) {
             throw e;
           }
-          // FIXME check that error still has index
-          // remove events up to the dup and retry
-          dupHashes.push(chunk[e.index].meta.eventHash);
-          chunk = chunk.slice(e.index + 1);
+          // if for some reason these are no writeErrors throw
+          if(!e.writeErrors) {
```

Comment:
This statement can be combined with the previous if clause.

Reply:
good catch and thanks for the comment I addressed it here:

https://github.com/digitalbazaar/bedrock-ledger-storage-mongodb/pull/53/commits/a17c2ee9a2295df60581d30e7ff0ff7df2f846c0